kimchi/cairo/src/runner/bytecode.rs

```diff
@@ -0,0 +1,106 @@
+// TODO(querolita):
+// - be able to index memory with any type (u256 in particular) not only usize
+
+use crate::runner::word::CairoWord;
+
+/// This data structure stores the memory of the program
+pub struct CairoBytecode {
+    /// length of the public memory
+    publen: u64,
+    /// full memory vector, None if non initialized
+    pub vect: Vec<Option<CairoWord>>,
+}
+
+impl CairoBytecode {
+    /// Create a new memory structure from a vector of u64
+    pub fn new(input: Vec<i128>) -> CairoBytecode {
+        // Initialized with the public memory (compiled instructions only)
+        // starts intentionally with a zero word for ease of testing
+        let mut aux = vec![0];
+        aux.extend(input);
+        CairoBytecode {
+            publen: (aux.len() - 1) as u64,
+            vect: aux.into_iter().map(|i| Some(CairoWord::new(i))).collect(),
+        }
+    }
+
+    /// Get size of the public memory
+    pub fn public(&self) -> u64 {
+        self.publen
+    }
+
+    /// Get size of the full memory including dummy 0th entry
+    pub fn size(&self) -> u64 {
```

Comment:
So the reason that it is returning u64 is that the code is working for writing and reading from memory data that fits into 64 bits. But it is in any case something that is going to change (aka the previous comment), as I am switching to field compatibility directly. Nonetheless, given the uses that this function may have, this one could return a usize, instead of u64 or field elements I think.